Python_Scripts/script_regroup_models.py

```diff
@@ -0,0 +1,337 @@
+            sbml_path = os.path.abspath(sbml_files[0])
+            model_info = _check_biomodels_model(sedml_model, sbml_path,
+                                                model_name, model_year,
+                                                model_info)
+
+    # Group sbml models and give them short Identifiers (e.g., Bachmann2011a)
+    model_info_df = _group_models_by_id(model_info)
+
+    return model_info_df
+
+
+def _group_models_by_id(model_info):
+    # We've run through all models. Now, let's generate short identifier for
+    # the groups of models, which should belong together
+    known_ids = {}
+    for model in model_info:
+        if model['long_id'] not in known_ids:
+            new_id = _generate_new_id(known_ids, model['long_id'])
+            known_ids[model['long_id']] = new_id
+        model['short_id'] = known_ids[model['long_id']]
+    # finally we want to post-process the model_list
+    return pd.DataFrame(model_info)
+
+
+def _generate_new_id(known_ids, long_id):
+    # count, which letter to append in case (yes, I know that's simplistic)
+    letter_count = 0
+    letter_list = ['', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i']
+    # iterate over the unique known_ids and increment counter
+    for known_id in known_ids:
+        if (known_id[0], known_id[1]) == (long_id[0], long_id[1]):
+            letter_count += 1
+    # append a letter, if necessary
+    append_letter = letter_list[letter_count]
+    # return the new short identifier
+    return str(long_id[0]) + str(long_id[1]) + append_letter
+
+
+def _parse_model_name(sedml_model):
+    # parse the sedml name via regex
+    model_name_full = sedml_model.split('_')[0]
+    p_name = re.compile('[A-Za-z]*')
+    p_year = re.compile('\d\d\d\d')
+    m_name = p_name.match(model_name_full)
+    m_year = p_year.search(model_name_full)
+
+    if m_name is None:
+        print('Model name could not be read. Model ' + sedml_model)
+        return None, None
+    else:
+        model_name = m_name[0].lower()
+
+    if m_year is None:
+        print('Model year could not be read. Model ' + sedml_model)
+        return None, None
+    else:
+        model_year = m_year[0]
+
+    return model_name, model_year
+
+
+def _check_biomodels_model(sedml_model, sbml_path, model_name, model_year, model_info):
+    # only one sbml file, a benchmark model on its own.
+    sedml_path = ''
+    sbml_model = (libsbml.readSBML(sbml_path)).getModel()
+    n_species = len(sbml_model.getListOfSpecies())
+    n_reactions = len(sbml_model.getListOfReactions())
+
+    # get the simulation times and write them to csv file
+    out_start, out_end, n_timepoints, _ = \
+        timePointsBioModels(sedml_model)
+    if out_start is None:
+        return model_info
+
+    model_info.append({
+        'name': model_name,
+        'year': model_year,
+        'n_species': n_species,
+        'n_reactions': n_reactions,
+        'long_id': (model_name, model_year, n_species, n_reactions),
```

Comment:
add documentation that "long_id" is the central variable defining what will be grouped? grouping looks reasonable to me. i.e. it will group models of the same name + year and same nr of species and reactions. thus it will allow to group the same model appearing in different figures, potentially simulated under different conditions, but still to be considered the same model.